clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/consumer/PlaintextConsumerFetchTest.java

```diff
@@ -0,0 +1,486 @@
+                consumer2,
+                tp2,
+                1,
+                24,
+                24,
+                startingTimestamp + 24 * hourMillis,
+                hourMillis
+            );
+
+            // seek to out of range position
+            outOfRangePos = totalRecords2 + 1;
+            consumer2.seek(tp2, outOfRangePos);
+            // assert that poll resets to the duration offset. consumer should read one record from last one hour
+            consumeAndVerifyRecords(
+                consumer2,
+                tp2,
+                1,
+                24,
+                24,
+                startingTimestamp + 24 * hourMillis,
+                hourMillis
+            );
+        }
+    }
+
+    @ClusterTest
+    public void testClassicConsumerFetchRecordLargerThanFetchMaxBytes() {
+        testFetchRecordLargerThanFetchMaxBytes(GroupProtocol.CLASSIC);
+    }
+
+    @ClusterTest
+    public void testAsyncConsumerFetchRecordLargerThanFetchMaxBytes() {
+        testFetchRecordLargerThanFetchMaxBytes(GroupProtocol.CONSUMER);
+    }
+
+    private void testFetchRecordLargerThanFetchMaxBytes(GroupProtocol groupProtocol) {
+        int maxFetchBytes = 10 * 1024;
+        checkLargeRecord(Map.of(
+            GROUP_PROTOCOL_CONFIG, groupProtocol.name().toLowerCase(Locale.ROOT),
+            FETCH_MAX_BYTES_CONFIG, maxFetchBytes
+        ), maxFetchBytes + 1);
+    }
+
+    @ClusterTest
+    public void testClassicConsumerFetchRecordLargerThanMaxPartitionFetchBytes() {
+        testFetchRecordLargerThanMaxPartitionFetchBytes(GroupProtocol.CLASSIC);
+    }
+
+    @ClusterTest
+    public void testAsyncConsumerFetchRecordLargerThanMaxPartitionFetchBytes() {
+        testFetchRecordLargerThanMaxPartitionFetchBytes(GroupProtocol.CONSUMER);
+    }
+
+    private void testFetchRecordLargerThanMaxPartitionFetchBytes(GroupProtocol groupProtocol) {
+        int maxFetchBytes = 10 * 1024;
+        checkLargeRecord(Map.of(
+            GROUP_PROTOCOL_CONFIG, groupProtocol.name().toLowerCase(Locale.ROOT),
+            MAX_PARTITION_FETCH_BYTES_CONFIG, maxFetchBytes
+        ), maxFetchBytes + 1);
+    }
+
+    private void checkLargeRecord(Map<String, Object> config, int producerRecordSize) {
+        try (Consumer<byte[], byte[]> consumer = cluster.consumer(config);
+             Producer<byte[], byte[]> producer = cluster.producer()
+        ) {
+            // produce a record that is larger than the configured fetch size
+            var record = new ProducerRecord<>(
+                tp.topic(),
+                tp.partition(),
+                "key".getBytes(),
+                new byte[producerRecordSize]
+            );
+            producer.send(record);
+
+            // consuming a record that is too large should succeed since KIP-74
+            consumer.assign(List.of(tp));
+            var records = consumer.poll(Duration.ofMillis(20000));
+            assertEquals(1, records.count());
```

Comment:
How about reusing `consumerRecords`?

```suggestion
            var records = consumeRecords(consumer, 1);
            assertEquals(1, records.size());
```